gateway-server/src/main/java/org/apache/knox/gateway/services/security/impl/ZookeeperRemoteAliasService.java

```diff
@@ -438,6 +457,9 @@ public void childEvent(final RemoteConfigurationRegistryClient client,
           client.removeEntryListener(path);
           if (paths.length > 1) {
             localAliasService.removeAliasForCluster(paths[0], paths[1]);
+            if (aliasCache.containsKey(paths[0])) {
+              aliasCache.get(paths[0]).remove(paths[1]);
+            }
```

Comment:
This could be another helper method for alias cache. Remove from aliasCache. Not sure if there is a possible race condition here between the check and remove.